src/main/java/mods/railcraft/common/blocks/machine/manipulator/TileFluidManipulator.java

```diff
@@ -113,11 +115,20 @@ protected void upkeep() {
         }
     }
 
+    @Override
+    public @Nullable EnumGui getGui() {
+        return EnumGui.MANIPULATOR_FLUID;
+    }
+
     @Override
     public boolean isItemValidForSlot(int slot, ItemStack stack) {
         if (slot == SLOT_INPUT) {
-            FluidStack filter;
-            return FluidItemHelper.isContainer(stack) && (FluidItemHelper.isEmptyContainer(stack) || (filter = getFilterFluid()) == null || FluidItemHelper.containsFluid(stack, filter));
+            boolean b1 = FluidItemHelper.isContainer(stack);
+            boolean b2 = FluidItemHelper.isEmptyContainer(stack);
+            FluidStack filter = getFilterFluid();
+            boolean b3 = filter == null;
+            boolean b4 = FluidItemHelper.containsFluid(stack, filter);
+            return b1 && (b2 || b3 || b4);
```

Comment:
This is less efficient because it looses the short circuit property of Boolean logic.

Reply:
Much easier to read though.

Reply:
I did that change during testing. Will revert.